submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/SubmissionFileSourceTest.kt

```diff
@@ -343,4 +341,52 @@ class SubmissionFileSourceTest(
             assertThat(fireFile.attributes).isEmpty()
         }
     }
+
+    @Test
+    fun `resubmission with SUBMISSION source ONLY`() {
+        val submission = tsv {
+            line("Submission", "S-FSTST7")
+            line("Title", "Submission Source Only")
+            line()
+
+            line("Study", "SECT-001")
+            line("Title", "Root Section")
+            line()
+
+            line("File", "test.txt")
+
+            line()
+        }.toString()
+
+        val file = tempFolder.createFile("test.txt", "test content")
+        webClient.uploadFiles(listOf(file))
+        val filesConfig = SubmissionFilesConfig(emptyList(), storageMode)
+        assertThat(webClient.submitSingle(submission, TSV, filesConfig)).isSuccessful()
+
+        val firstVersion = submissionRepository.getExtByAccNo("S-FSTST7")
+        val subFilesPath = "$submissionPath/${firstVersion.relPath}/Files"
+        val innerFile = Paths.get("$subFilesPath/test.txt")
+
+        assertThat(innerFile).exists()
+        assertThat(innerFile.toFile().readText()).isEqualTo("test content")
+
+        webClient.uploadFiles(listOf(tempFolder.createFile("test.txt", "updated test content")))
+
+        assertThat(
+            webClient.submitSingle(
+                submission,
+                TSV,
+                SubmissionFilesConfig(emptyList(), storageMode, preferredSources = listOf(SUBMISSION)),
+            )
+        ).isSuccessful()
+        assertThat(innerFile.toFile().readText()).isEqualTo("test content")
+
+        if (enableFire) {
```

Comment:
we can validate the same for NFS that the NFS file is used and not the user folder one.

Reply:
That's the content validation in line 382. This validation is to check that actually the same fire object, with the same key, was reused instead of creating a new object with the same content.